pkg/handlers/archive.go

```diff
@@ -115,6 +119,31 @@ func (a *Archive) openArchive(ctx logContext.Context, depth int, reader io.Reade
 }
 
 func (a *Archive) handleNonArchiveContent(ctx logContext.Context, reader io.Reader, archiveChan chan []byte) error {
+	buffer := make([]byte, 512)
+	n, err := reader.Read(buffer)
+	if err != nil && !errors.Is(err, io.EOF) {
+		return fmt.Errorf("unable to read file for MIME type detection: %w", err)
+	}
+
+	// Create a new reader that starts with the buffer we just read
+	// and continues with the rest of the original reader.
+	reader = io.MultiReader(bytes.NewReader(buffer[:n]), reader)
```

Comment:
Would using [bufio.Peek](https://pkg.go.dev/bufio#Reader.Peek) be equivalent here? I find that a little easier to understand than reading a buffer then resetting it with a multireader.

```go
bufReader := bufio.NewReaderSize(reader, 512)
buffer, err := bufReader.Peek(512)
if err != nil && !errors.Is(err, io.EOF) {
    // ...
}
```

Reply:
beautiful, thank you!

Reply:
Ahh weird... with that change the tests fail. I'll dig into why.

Reply:
Hm.. did you replace all the `reader` variables with `bufReader`? That was the only downside I could think of with this approach.

Reply:
```
// Peek returns the next n bytes without advancing the reader. The bytes stop
// being valid at the next read call
```

Reply:
I take that to mean the slice returned by `Peek` becomes invalid (since it's probably returning the internal buffer to avoid a copy). Do we need the buffer to remain valid after the next read?